src/executor/execute.rs

```diff
@@ -45,9 +46,20 @@ pub fn execute<T: 'static + Debug, U: Store<T> + StoreMut<T>>(
     };
 
     match prepared {
-        Prepared::Create(schema) => storage
-            .insert_schema(&schema)
-            .map(|(storage, _)| (storage, Payload::Create)),
+        Prepared::Create {
+            schema,
+            if_not_exists,
+        } => match storage.insert_schema(&schema) {
+            Ok((storage, _)) => Ok((storage, Payload::Create)),
+            Err((storage, Error::Store(StoreError::SchemaAlreadyExists))) => {
+                if if_not_exists {
+                    Ok((storage, Payload::Create))
```

Comment:
@leoppro 
Which I mentioned before, "Terminal" and "Non-terminal" were something more conceptual.
For now, what we need to see is simple.
On `match storage.insert_schema(&schema)`, though it returns error with `Err((storage, Error::Store(StoreError::SchemaAlreadyExists)))` but it transforms into `Ok((storage, Payload::Create))` on the specific condition `if if_not_exists`.

Once an error occurred, it's ok to change into another error message but, not good to change into non-error.

What I requested for the first one "Error is handled as proper business logic",
you can simply think as removing work of `Err` -> `Ok` cases.

Reply:
@panarch I understand what you are suggesting is this: I need to return another error and let the business logic layer decide whether to ignore that error. When `if_not_exists` is true and the schema already exists, I should return an error like `SchemaAlreadyExistsButCanIgnore`, then let the business logic layer(which is also the user) handle the error.

Reply:
I think there are some drawbacks to this: 
1. It's more unique and behaves differently from other similar databases, such as sqlite and MySQL. 
2. For the ORM framework, in order to adapt to gluesql, they need to implement special error handling logic.

Reply:
To solve this problem, I suggest adding a warning output to the `Glue::execute`, some errors that can be ignored are output to the user by warnings

Reply:
@leoppro 
Ok, now I think I also got what you're not sure about.
At first, 
> I need to return another error and let the business logic layer decide whether to ignore that error.
My description wasn't enough, what I wanted to say was which "business logic layer decide whether to ignore that error" should not happen.
Once error occurs, then it's an error, it should not be ignored or accepted.

Current nested 2-level error matches can be reduced into one.
Like this,
* Replace `StoreError::SchemaAlreadyExists` to `ExecuteError::SchemaAlreadyExists`
* Revert `sled_storage.rs` changes and move that business logic codes to `execute.rs`
It means, `execute.rs` will check whether schema exists or not. (not in `storage`)

How do you think?

Reply:
> It's more unique and behaves differently from other similar databases, such as sqlite and MySQL.
Your code changes are already good enough, I don't think we need to change behaviors which affects to users.
My opinion is only about internal implementation, not about outer behavior.